.ci/build-kit/compile.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -e
+
+# Install coverage dependencies
+apt update && apt install -y gcovr lcov
+
+# Install Rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+source "$HOME/.cargo/env"
```

Comment:
I'm not sure on what the best solution here is.  Ideally all the build dependencies are already setup in the build container, but this would mean we either have one large container, which contains all the dependencies, or multiple (probably a lot) smaller ones, the contain only the dependencies needed.

Reply:
I think we can figure it out once we have more stability